src/main/scala/io/kf/etl/context/DefaultContext.scala

```diff
@@ -46,14 +47,23 @@ class DefaultContext extends AutoCloseable {
   private def initSparkSession(withES: Boolean): SparkSession = {
     val session = SparkSession.builder().appName("Kids First Portal ETL")
 
+    val user = getOptionalConfig(CONFIG_NAME_ES_USER, configMutable)
+    val pwd = getOptionalConfig(CONFIG_NAME_ES_PASS, configMutable)
+
     if (withES) {
-      session
-        .config("es.nodes", elasticSearchUrl(configMutable))
-        .config("es.nodes.wan.only", Option(configMutable.getBoolean(CONFIG_NAME_ES_NODES_WAN_ONLY)).getOrElse(false))
-        .getOrCreate()
-    } else {
-      session.getOrCreate()
+      if (user.isDefined && pwd.isDefined) {
+        session
+          .config("es.nodes", elasticSearchUrl(configMutable))
+          .config("es.nodes.wan.only", Option(configMutable.getBoolean(CONFIG_NAME_ES_NODES_WAN_ONLY)).getOrElse(false))
+          .config("es.net.http.auth.user", user.get)
+          .config("es.net.http.auth.pass", pwd.get)
+      } else {
+        session
+          .config("es.nodes", elasticSearchUrl(configMutable))
+          .config("es.nodes.wan.only", Option(configMutable.getBoolean(CONFIG_NAME_ES_NODES_WAN_ONLY)).getOrElse(false))
+      }
     }
+    session.getOrCreate()
```

Comment:
It will create an emty session (initiated L48) but it will not considered config setu that is been done between line 54 and 64